### ISSUE TITLE
SKCanvas native memory leak

### ISSUE THREAD

**Issue by mattleibow** [mono/SkiaSharp#192]
There appears to be a memory leak when referencing `SKCanvas`. 

> Once you create an `SKCanvas`, I could not find any way to release memory used by it.
> The following lines create a memory leak of 150 megs, attached is the allocations screenshot.

```csharp
for (int i = 0; i < 10; i++) {
    using (SkiaSharp.SKBitmap bitmap = new SkiaSharp.SKBitmap(2000, 2000, false)) {
        using (var canvas = new SkiaSharp.SKCanvas(bitmap)) {

        }
    }
}
```
> The following releases memory without issues.

```csharp
for (int i = 0; i < 10; i++) {
    using (SkiaSharp.SKBitmap bitmap = new SkiaSharp.SKBitmap(2000, 2000, false)) {

    }
}
```

> I have tried, flush, dispose, setting canvas to null, none of them are working. Basically, I'm trying to draw paths and save them as PNG.
> Any ideas?

The original forum post: https://forums.xamarin.com/discussion/82206/skiasharp-skimage-encode-causes-memory-leaks

**Comment by mattleibow** [comment]
I think that this is a result of not dereferencing the underlying native canvas object.

**Comment by mattleibow** [comment]
I fixed this in 7d7e97463bba60790d25cbc616dd85a59a5d59dd

Typically a canvas is owned by the object that provided it, for example a surface, picture recorder or document. But in one special situation, it can be constructed from a bitmap.
